src/components/Todo.jsx

```diff
@@ -0,0 +1,260 @@
+import { useState } from 'react'
+
+const Todo = () => {
+
+    // states
+    const [tasks, setTasks] = useState([])
+    const [userTextInputValue, setUserTextInputValue] = useState("")
+    const [todoListName, setTodoListName] = useState("My Todo List")
+    const [todoListNameHolder, setTodoListNameHolder] = useState("")
+    const [isFiltered, setIsFiltered] = useState(false)
+    const [preFilteredList, setPreFilteredList] = useState([])
+    const [showTaskListItems, setShowTaskListItems] = useState(false)
+
+    const setTaskListLengthText = function () {
+        let list = []
+        if (!isFiltered) {
+            list = tasks
+        }
+        else {
+            list = preFilteredList
+        }
+        if (showTaskListItems) {
+            if (list.length === 1) {
+                return(`(${list.length} Item)`)
+            }
+            else if (list.length === 0) {
+                return(``)
+            }
+            else {
+                return(`(${list.length} Items)`)
+            }
+        }
+        else {
+            return(``)
+        }
+    }
+
+    const sortTodoList = function (type, e) {
+        e.preventDefault(e);
+        let filteredList = tasks
+        if (type === "ABC") {
+            filteredList.sort(((a, b) => a.title.localeCompare(b.title)))
+            setTasks([...filteredList])
+        }
+        else if (type === "Time") {
+            filteredList.sort(((a, b) => a.createdAt > b.createdAt))
+            setTasks([...filteredList])
+        }
+    }
+
+    const filterTodoList = function (title, e) {
+        e.preventDefault(e);
+        if (userTextInputValue !== "") {
+            if (todoListName === "Please enter a name for your item") {
+                setShowTaskListItems(false)
+                setTodoListName(todoListNameHolder)
+            }
+        }
+        let filteredList = []
+        tasks.map(item => {
+
+            if (isFiltered === false) {
```

Comment:
You might want to make this less verbose using the bang operator e.g. `!isFiltered`